seerep_srv/seerep_core/src/core_dataset.cpp

```diff
@@ -448,14 +497,29 @@ CoreDataset::getAllDatasetUuids(std::shared_ptr<seerep_core::CoreDataset::Dataty
 
   datatypeSpecifics->timetree.query(boost::geometry::index::intersects(aabbtime),
                                     std::back_inserter(allIdsFromTimeTree));
-  std::vector<boost::uuids::uuid> allIds;
-  for (auto it = std::make_move_iterator(allIdsFromTimeTree.begin()),
-            end = std::make_move_iterator(allIdsFromTimeTree.end());
-       it != end; ++it)
+
+  if (sortByTime)
   {
-    allIds.push_back(std::move(it->second));
+    std::vector<std::pair<int64_t, boost::uuids::uuid>> sortingVector;
+    for (auto it = std::make_move_iterator(allIdsFromTimeTree.begin()),
+              end = std::make_move_iterator(allIdsFromTimeTree.end());
+         it != end; ++it)
+    {
+      sortingVector.push_back(std::make_pair(boost::geometry::get<0>(it->first.min_corner()), it->second));
+    }
+    return sortResultByTimeActualSorting(sortingVector);
```

Comment:
Please reuse the code from [sortResultByTime](https://github.com/agri-gaia/seerep/blob/f0c986de31ff366c70b2b698474e7dd30d0030f6/seerep_srv/seerep_core/src/core_dataset.cpp#L427)